### PR TITLE
[Fix] Substitute SyncBN with BN for mmpose

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

All other codebases can substitute SyncBN with BN in `build_pytorch_model` function from the `BaseTask` class, while mmpose overwrites the `build_pytorch_model` function without substitute SyncBN with BN. It will failed on RTMPose deployment. 

## Modification

Call the mmengine API `revert_sync_batchnorm` in mmpose `build_pytorch_model` function.

## BC-breaking (Optional)

None.

## Use cases (Optional)

RTMPose deployment.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by grimoire — APPROVED** [review]

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by RunningLeon — COMMENTED** [review]
ci failed, could fix it before merge?

**Comment by lvhan028** [comment]
Why cannot reuse BaseTask's `build_pytorch_model`?
Doesn't mmpose support `mmengine.registry.MODELS.build(model)`?

**Review by RunningLeon — COMMENTED** [review]
pls check this change with regression test

File: mmdeploy/codebase/mmpose/deploy/pose_detection.py [hunk 1]
```diff
@@ -145,31 +145,8 @@ class PoseDetection(BaseTask):
     def __init__(self, model_cfg: mmengine.Config, deploy_cfg: mmengine.Config,
                  device: str):
         super().__init__(model_cfg, deploy_cfg, device)
-
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmpose/deploy/pose_detection.py:148]: CI failed, you may need to sync with latest `dev-1.x` branch

**Pull request merged**